Modules/QuestieInit.lua

```diff
@@ -311,30 +344,31 @@ end
 -- this function creates coroutine to run a function from QuestieInit.Stages[]
 ---@param stage number @the stage to start
 function _QuestieInit:StartStageCoroutine(stage)
-    local initFrame = CreateFrame("Frame")
-    local routine = coroutine.create(QuestieInit.Stages[stage])
-
-    local function InitOnUpdate()
-        local success, ret = coroutine.resume(routine)
-        if success then
-            if coroutine.status(routine) == "dead" then
-                initFrame:SetScript("OnUpdate", nil)
-                initFrame:SetParent(nil)
-                initFrame = nil
-                if type(ret) == "function" then -- continue to next stage, if it was returned by coroutine
-                    ret()
-                end
-            end
-        else
-            Questie:Error(l10n("Error during initialization!"), ret)
-            initFrame:SetScript("OnUpdate", nil)
-            initFrame:SetParent(nil)
-            initFrame = nil
+    local coStatus, coResume = coroutine.status, coroutine.resume
```

Comment:
This could also be done outside of the function for even better performance, right?